matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/create/CreateLocalRoomTask.kt

```diff
@@ -228,11 +237,40 @@ internal class DefaultCreateLocalRoomTask @Inject constructor(
             )
         }
 
+        val localRoomThreePidEvents = invited3Pids.map { body ->
```

Comment:
Maybe this could be extracted in 2 private methods? And I am struggling to understand the need for both `LocalRoomThirdPartyInviteContent` and `RoomThirdPartyInviteContent`. Since we are creating a local room, I thought we would only have `LocalRoomThirdPartyInviteContent`. Maybe this could be explained through methods names or code comments.

Reply:
I created a `LocalRoomThirdPartyInviteContent` because I had to persist the full `ThreePid` to propagate it to create the "real" room. When we invite someone by email to a room, the email address is obfuscated in the `RoomThirdPartyInviteContent` and there is no way to retrieve it.
I created a new object for this purpose because I would not modify the existing object behaviour.
I also need the "RoomThirdPartyInviteContent" for the display of the timeline of the local room which should be as close as possible to a real room.
That's why I have both objects, I'll see to clarify this part.

Reply:
Update: I don't need it anymore the local event for the purpose of the room creation but it is still necessary for the correct display of the local timeline. I can remove it if we agree that the obfuscated three pid is enough.